### PR TITLE
Improve ESLint configuration, and fix violations.

### PR DESCRIPTION
**Pull request by jasedit**
This PR more extensively configures ESLint, including pointing it at the files in `src`, causing it to identify a number of issues. This PR also starts trying to clean those up, though is still a work in progress.

### CODE REVIEW

File: src/macros/macros.js [hunk 1]
```diff
@@ -39,9 +40,9 @@ module.exports = class Macros {
       // console.log(e)
     })
 
-    KeyboardJS.on("right", (e) => {
+    KeyboardJS.on("right", (_) => {
       document.dispatchEvent(
-        new Event("autocomplete/right")
+        new Events.Event("autocomplete/right")
```
Comment (ondreian) [inline src/macros/macros.js:45]: This is not correct, we are using a browser [`Event`](https://developer.mozilla.org/en-US/docs/Web/Events)
not an EventEmitter event

**Comment by ondreian** [comment]
I know it's just a draft, but I wanted to point out some mistakes I assume `eslint` suggested fixes for here that were actually not mistakes, Electron is both `node` and `browser` context

**Comment by jasedit** [comment]
That's why I put this up as a draft! I'll work on cleaning up those issues and go from there. This still has other failures present, so there's more work to be done yet.

**Comment by jasedit** [comment]
Flipping both flags to true appears to resolve a number of these issues, making this a lot easier to wrap up. One question - do you think ESLint should run in CI? Right now it's only a pre-commit hook, but that can be bypassed if someone cares. I can try and configure it in CI or not as you prefer.

**Comment by ondreian** [comment]
I think it should be run in CI, but that should be a separate followup PR and I will likely need to enable Travis for this project

File: src/macros/macros.js [hunk 2]
```diff
@@ -51,7 +51,7 @@ module.exports = class Macros {
         e.preventDefault()
         const sess = Session.list()[idx]
         if (sess && sess !== Session.focused()) {
-          Bus.emit(Bus.events.FOCUS, sess)
+          Bus.emit(Bus.FOCUS, sess)
```
Comment (ondreian) [inline src/macros/macros.js:54]: This is not correct https://github.com/elanthia-online/illthorn/blob/d47c2426cf794429140ec46ecb59b5cc363ef393/src/bus/index.js#L5

File: src/vimish/commands/connect.js [hunk 3]
```diff
@@ -1,13 +1,8 @@
 const Command = require("../command")
-const Bus = require("../../bus")
 const Autodetect = require("../../autodetect")
+const redraw = require("../../util/redraw")
```
Comment (ondreian) [inline src/vimish/commands/connect.js:3]: this needs to be a destructed argument: `{redraw}` since you are namespacing it in the redraw util

Comment (jasedit) [reply src/vimish/commands/connect.js:3]: Is that the correct term for this syntax? It looks like this is declaring a const set, but that's drawing from other languages, so I don't think this is the right read. What term should I look up to better understand how to pull things from a namespaced module?

File: src/macros/macros.js (hunk 2 above)
Comment (jasedit) [reply src/macros/macros.js:54]: Do you mean the change is not correct, or the event listed is not correct?

Comment (ondreian) [reply src/macros/macros.js]: It should be `events.FOCUS`, there is no `Bus.focus`

File: src/vimish/commands/connect.js (hunk 3 above)
Comment (ondreian) [reply src/vimish/commands/connect.js]: Yeah, here is a good over-view for you [Destructuring Assignment](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment)

File: src/macros/macros.js (hunk 2 above)
Comment (jasedit) [reply src/macros/macros.js]: I made a guess that's what you meant, so this should already be fixed.

**Review by ondreian — APPROVED** [review]
seems good, thanks!

**Pull request merged**